javascript/lib/api/src/options/request.options.ts

```diff
@@ -139,7 +146,12 @@ interface HasFilterAndNamespace<T extends HasFilterAndNamespace<T>> {
   withNamespaces(...namespaces: string[]): T;
 }
 
-abstract class AbstractRequestOptionsWithMatchOptions<T extends AbstractRequestOptionsWithMatchOptions<T>>
+export interface RequestOptionsWithMatchOptions<T extends RequestOptionsWithMatchOptions<T>>
+  extends AddRequestOptions<RequestOptionsWithMatchOptions<T>>,
+  HasMatch<RequestOptionsWithMatchOptions<T>> { }
+
+
+export abstract class AbstractRequestOptionsWithMatchOptions<T extends AbstractRequestOptionsWithMatchOptions<T>>
   extends AbstractRequestOptions<AbstractRequestOptionsWithMatchOptions<T>>
   implements HasMatch<AbstractRequestOptionsWithMatchOptions<T>> {
```

Comment:
We should implement `RequestOptionsWithMatchOptions` instead of `HasMatch` here

Reply:
Thanks, fixed it already :)